src/main/java/mrjake/aunis/tileentity/stargate/StargateAbstractBaseTile.java

```diff
@@ -693,6 +673,21 @@ public void onLoad() {
 			AunisPacketHandler.INSTANCE.sendToServer(new StateUpdateRequestToServer(pos, StateTypeEnum.RENDERER_STATE));
 		}
 	}
+
+	public void generateAddresses(boolean reset) {
+		Random random = new Random(pos.hashCode() * 31 + world.provider.getDimension());
+
+		for (SymbolTypeEnum symbolType : SymbolTypeEnum.values()) {
+			StargateAddress address = getStargateAddress(symbolType);
+
+			if (gateAddressMap.get(symbolType) == null || reset) {
+				address = new StargateAddress(symbolType);
+				address.generate(random);
+			}
+
+			this.setGateAddress(symbolType, address);
```

Comment:
Address might be `null`

Reply:
Also there is so much code stripped of this method compared to the original code from `onLoad`

Reply:
Yeah beacause it was basically `setGateAddress` repeated.